### PR TITLE
RUMM-3470: Update documentation for Android SDK v2

### PR DESCRIPTION
**Pull request by 0xnm**
### What does this PR do?

This PR updates documentation for the features provided by the [Datadog Android SDK](https://github.com/DataDog/dd-sdk-android) in accordance with the changes made with upcoming SDK v2 release.

With SDK v2 we split the `dd-sdk-android` monolith into different parts:

* `dd-sdk-android-core` - core module, containing the code to handle common parts of SDK (data storage, data upload), etc.
* `dd-sdk-android-rum` - module containing RUM feature functionality
* `dd-sdk-android-logs` - module containing Logs feature functionality
* `dd-sdk-android-trace` - module containing Trace feature functionality
* `dd-sdk-android-webview` - module containing WebView Tracking functionality
* `dd-sdk-android-okhttp` - module containing OkHttp instrumentation functionality

This split allows to use and configure features independently.

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by xgouchet — APPROVED** [review]

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 1]
```diff
@@ -30,9 +30,13 @@ If you use Glide to load images in your application, take a look at Datadog's [d
 
 If you use Jetpack Compose in your application, take a look at Datadog's [dedicated Jetpack Compose library][7].
 
+### RxJava
+
+If you use RxJava in your application, take a look at Datadog's [dedicated RxJava library][8].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:35]:
```suggestion
If you use RxJava in your application, see Datadog's [dedicated RxJava library][8].
```

File: content/en/real_user_monitoring/android/troubleshooting.md [hunk 2]
```diff
@@ -50,44 +50,12 @@ Each batch follows the intake specification. They are sent as soon as the networ
...
-
-
-### Attributes
-
-In earlier versions, attributes were created or removed with the `Logger.addField()` or `Logger.removeField()`
-methods. These methods were renamed for consistency purposes, and are now `Logger.addAttribute()`
- and `Logger.removeAttribute()`. Their behavior remains the same.
+## Migrating to 2.0.0
 
+If you've been using the SDK v1, there are some breaking changes introduced in version `2.0.0`. Check our [migration guide][1] to get more information.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/troubleshooting.md:55]:
```suggestion
If you've been using the SDK v1, there are some breaking changes introduced in version `2.0.0`. See the [migration guide][1] for more information.
```

File: content/en/real_user_monitoring/android/web_view_tracking.md [hunk 3]
```diff
@@ -30,31 +30,25 @@ Set up the web page you want rendered on your mobile Android and Android TV appl
 
 ### Update your existing SDK setup
 
-1. Download the [latest version][2] of the RUM Android SDK.
-2. Edit your existing Android SDK setup from [RUM Android Monitoring][3].
-3. Add tracking for web views with the following example:
+1. If you want to forward RUM events coming from web page, download the [latest version][2] of RUM Android SDK and setup RUM feature following the [dedicated guide][3].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/web_view_tracking.md:33]:
```suggestion
1. If you want to forward RUM events coming from web pages, download the [latest version][2] of RUM Android SDK and setup RUM feature following the [dedicated guide][3].
```

File: content/en/real_user_monitoring/android/web_view_tracking.md [hunk 4]
```diff
@@ -30,31 +30,25 @@ Set up the web page you want rendered on your mobile Android and Android TV appl
...
+2. If you want to forward Log events coming from web page, download the [latest version][4] of Logs Android SDK and setup Logs feature following the [dedicated guide][5].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/web_view_tracking.md:34]:
```suggestion
2. If you want to forward Log events coming from web pages, download the [latest version][4] of Logs Android SDK and setup Logs feature following the [dedicated guide][5].
```

File: content/en/real_user_monitoring/android/web_view_tracking.md [hunk 5]
```diff
@@ -30,31 +30,25 @@ Set up the web page you want rendered on your mobile Android and Android TV appl
...
-        }
+    ```groovy
+    dependencies {
+        implementation "com.datadoghq:dd-sdk-android-webview:x.x.x"
     }
-   ```
+    ```
 
-4. Configure the `DatadogEventBridge` for web views you want to track in your mobile Android application using `DatadogEventBridge.setup(webView)` in the configuration file when you initialize the RUM Android SDK.
+4. Enable tracking for web view you want with the following code snippet:
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/web_view_tracking.md:43]:
```suggestion
4. Enable tracking for web views with the following code snippet:
```

File: content/en/real_user_monitoring/error_tracking/android.md [hunk 6]
```diff
@@ -135,7 +135,7 @@ This resolves the final value for the `versionName` property as `fooBar`.
...
 | `versionName`              | The version name of the application (by default, the version declared in the `android` block of your `build.gradle` script).                                                                                                               |
 | `serviceName`              | The service name of the application (by default, the package name of your application as declared in the `android` block of your `build.gradle` script).                                                                                                                          |
-| `site`                     | The Datadog site to upload your data to (US1, US3, US5, EU1, or US1_FED).                                                                                                                                       |
+| `site`                     | The Datadog site to upload your data to (US1, US3, US5, EU1, US1_FED or AP1).                                                                                                                                       |
```
Comment (rtrieu) [inline content/en/real_user_monitoring/error_tracking/android.md:138]:
```suggestion
| `site`                     | The Datadog site to upload your data to (US1, US3, US5, EU1, US1_FED, or AP1).                                                                                                                                       |
```

File: content/en/real_user_monitoring/error_tracking/reactnative.md [hunk 7]
```diff
@@ -368,11 +368,11 @@ If you are still unsure, you can see if running `(cd android && ./gradlew tasks
 
 #### Manually on each build
 
-In your `android/app/build.gradle` file, add the plugin and configure it **at the very top of the file**:
+In your `android/app/build.gradle` file, add the [latest version of plugin][15] and configure it **at the very top of the file**:
```
Comment (rtrieu) [inline content/en/real_user_monitoring/error_tracking/reactnative.md:371]:
```suggestion
In your `android/app/build.gradle` file, add the [latest version of the plugin][15] and configure it **at the top of the file**:
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 8]
```diff
@@ -30,9 +30,13 @@ If you use Glide to load images in your application, take a look at Datadog's [d
...
+
 ### Picasso
  
-If you use Picasso, let it use your `OkHttpClient` for RUM and APM information about network requests made by Picasso.
+If you use Picasso, let it use your `OkHttpClient` instrumented with Datadog SDK for RUM and APM information about network requests made by Picasso.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:39]:
```suggestion
If you use Picasso, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about network requests made by Picasso.
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 9]
```diff
@@ -57,7 +61,7 @@ If you use Picasso, let it use your `OkHttpClient` for RUM and APM information a
  
 ### Retrofit
  
-If you use Retrofit, let it use your `OkHttpClient` for RUM and APM information about network requests made with Retrofit.
+If you use Retrofit, let it use your `OkHttpClient` instrumented with Datadog SDK for RUM and APM information about network requests made with Retrofit.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:64]:
```suggestion
If you use Retrofit, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about network requests made with Retrofit.
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 10]
```diff
@@ -80,7 +84,7 @@ If you use Retrofit, let it use your `OkHttpClient` for RUM and APM information
  
 ### SQLDelight
  
-If you use SQLDelight, take a look at Datadog's [dedicated SQLDelight library][4].
+If you use SQLDelight in your application, take a look at Datadog's [dedicated SQLDelight library][4].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:87]:
```suggestion
If you use SQLDelight in your application, see Datadog's [dedicated SQLDelight library][4].
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 11]
```diff
@@ -121,20 +125,20 @@ RUM error event for it.
  
 ### Apollo (GraphQL)
  
-If you use Apollo, let it use your `OkHttpClient` for RUM and APM information about all the queries performed through Apollo client.
+If you use Apollo, let it use your `OkHttpClient` instrumented with Datadog SDK for RUM and APM information about all the queries performed through Apollo client.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:128]:
```suggestion
If you use Apollo, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about all the queries performed through Apollo client.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some suggestions for consistency with our style guide, but otherwise looks good!

**Review by rtrieu — APPROVED** [review]
Looks good!

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 12]
```diff
@@ -30,9 +30,13 @@ If you use Glide to load images in your application, take a look at Datadog's [d
...
 If you use Jetpack Compose in your application, take a look at Datadog's [dedicated Jetpack Compose library][7].
 
+### RxJava
+
+If you use RxJava in your application, see Datadog's [dedicated RxJava library][8].
+
 ### Picasso
  
-If you use Picasso, let it use your `OkHttpClient` for RUM and APM information about network requests made by Picasso.
+If you use Picasso, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about network requests made by Picasso.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:39]:
```suggestion
If you use Picasso, use it with the `OkHttpClient` that's been instrumented with the Datadog SDK for RUM and APM information about network requests made by Picasso.
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 13]
```diff
@@ -57,7 +61,7 @@ If you use Picasso, let it use your `OkHttpClient` for RUM and APM information a
  
 ### Retrofit
  
-If you use Retrofit, let it use your `OkHttpClient` for RUM and APM information about network requests made with Retrofit.
+If you use Retrofit, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about network requests made with Retrofit.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:64]:
```suggestion
If you use Retrofit, use it with the `OkHttpClient` that's been instrumented with the Datadog SDK for RUM and APM information about network requests made with Retrofit.
```

File: content/en/real_user_monitoring/android/integrated_libraries.md [hunk 14]
```diff
@@ -121,20 +125,20 @@ RUM error event for it.
  
 ### Apollo (GraphQL)
  
-If you use Apollo, let it use your `OkHttpClient` for RUM and APM information about all the queries performed through Apollo client.
+If you use Apollo, use it with the `OkHttpClient` that's been instrumented with Datadog SDK for RUM and APM information about all the queries performed through Apollo client.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/android/integrated_libraries.md:128]:
```suggestion
If you use Apollo, use it with the `OkHttpClient` that's been instrumented with the Datadog SDK for RUM and APM information about all the queries performed through Apollo client.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Sorry just noticed a few more things.

**Comment by 0xnm** [comment]
@rtrieu I've addressed your comments and also added 2 new commit with additional edits, can you please review this PR again?

**Review by rtrieu — APPROVED** [review]
LGTM

**Comment by rtrieu** [comment]
@0xnm Let me know when this is ready to merge!

**Pull request merged**